File: Prácticas1DAW/ExcepcionesJava/Gato.java
package excepcionesJava;
/**
 * Clase Gato
 * @author juanbu97
 *
 */
public class Gato {

	private String sexo;
	private String color;
	private String raza;
	
	public Gato(String sexo) {
		setSexo(sexo);
	}
	
	public String getSexo() {
		return sexo;
	}

	private void setSexo(String sexo) {
		this.sexo = sexo;
	}

	public void maulla() {
		System.out.println("Miau");
	}
	
	public Gato apareaCon(Gato gato2) throws ApareamientoImposible{
		if(this.getSexo().equals(gato2.getSexo())) {
			throw new ApareamientoImposible();
		}
			String hijo = (int)(Math.random() * 2) ==  0 ? "hembra" : "macho";
			return new Gato(hijo);
		
	}

	@Override
	public String toString() {
		return "Gato [sexo=" + sexo + "]\n";
	}

	
	
}

File: Prácticas1DAW/EjercicioPoo1/BujalanceGarciaJuanAntonioObjetos/Python/Gato.py
'''
Created on 11 ene. 2019

@author: d18bugaj
'''

class Gato(object):
    '''
    classdocs
    '''


    def __init__(self, sexo):
        self.sexo = sexo
        self.color = ""
        self.raza = ""
        self.edad = 0
        self.peso = 0
        
    def get_sexo(self):
        return self.sexo
    
    def maulla(self):
        print("Miau")
        
    def ronrronea(self):
        print("Rrrrrrr")
        
        
    def come(self, comida):
        if comida == "pescado":
            print("Que rico")
        else:
            print("Yo solo como pescado")
            
    
    def pelea_con(self,contrincante):
        if self.sexo == "hembra":
            print("No me gusta pelear")
        else:      
            if contrincante.sexo == "hembra":
                print("No peleo con gatitas")
            else:
                print("Ven aqui que te vas a enterar")
                
#Prueba de la clase
if __name__ == "__main__":
    floro = Gato("macho")
    floro.maulla()
    floro.come("pienso")
    floro.come("pescado")
                
                
    florinda = Gato("hembra")
    florinda.pelea_con(floro)
    
    pulgoso = Gato("macho")
    pulgoso.pelea_con(floro)
    
    pulgoso.pelea_con(florinda)

        
        